### PR TITLE
Created TUICheckBox and TUIToggleRow

### PR DESCRIPTION
**Pull request by arvindhsukumar**

### CODE REVIEW

File: Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+/// A row with a content view and a toggle switch
+/// - Parameters:
+/// - contentView: The content view to be displayed
+/// - toggle: The toggle switch to be displayed
+struct TUIToggleRow<Content: View>: View {
+  public var contentView: () -> Content
+  public var toggle: () -> TUIToggleSwitch
+  
+  var body: some View {
```
Comment (maheshwaran01m) [inline Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift:18]:
```suggestion
  public var body: some View {
```

**Review by maheshwaran01m — APPROVED** [review]

File: Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    HStack {
```
Comment (gopinathcholan) [inline Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift:19]:
```suggestion
    HStack(spacing: 0) {
```

File: Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift (hunk 1 above)
Comment (arvindhsukumar) [reply Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift]: Updated

File: Sources/TarkaUI/Components/Base Components/CheckBox/TUICheckBox.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//
+//
+//  Created by Arvindh Sukumar on 21/02/24.
+//
+
+import SwiftUI
+
+public struct TUICheckBox: View {
+  var isSelected: Bool = false
+
```
Comment (maheshwaran01m) [inline Sources/TarkaUI/Components/Base Components/CheckBox/TUICheckBox.swift:12]:
```
public init(isSelected: Bool) {
self.isSelected = isSelected
}
```

File: Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+/// - contentView: The content view to be displayed
+/// - toggle: The toggle switch to be displayed
+public struct TUIToggleRow<Content: View>: View {
+  public var contentView: () -> Content
+  public var toggle: () -> TUIToggleSwitch
+  
+  public var body: some View {
+    HStack(spacing: Spacing.baseHorizontal) {
+      contentView()
+      Spacer()
```
Comment (maheshwaran01m) [inline Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift:21]:
```suggestion
   .frame(maxWidth: .infinity, alignment: .leading)
```

File: Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import SwiftUI
+
+/// A row with a content view and a toggle switch
+/// - Parameters:
+/// - contentView: The content view to be displayed
+/// - toggle: The toggle switch to be displayed
+public struct TUIToggleRow<Content: View>: View {
+  public var contentView: () -> Content
+  public var toggle: () -> TUIToggleSwitch
+
```
Comment (maheshwaran01m) [inline Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift:17]:
```
public init(toggle: @escaping () -> TUIToggleSwitch, @ViewBuilder content: @escaping () -> Content {
 self.toggle = toggle
 self.content: content
}
```

File: Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift (hunk 4 above)
Comment (arvindhsukumar) [reply Sources/TarkaUI/Components/ToggleRow/TUIToggleRow.swift:21]: Spacer() is needed to ensure that the toggle view is right-aligned

**Pull request merged**